### PR TITLE
2.0-preview branch

### PR DESCRIPTION
**Pull request by bicknellr**

### CODE REVIEW

File: demo/simple-menu.html [hunk 1]
```diff
@@ -15,15 +15,17 @@
 <dom-module id="simple-menu">
   <template>
     <style>
-      .content ::content > .iron-selected {
+      ::slotted(*) {
```
Comment (e111077) [inline demo/simple-menu.html]: Why does this have to be added now? Is this standard or for an older version of Polymer?

File: test/iron-menu-behavior.html [hunk 2]
```diff
@@ -139,107 +139,129 @@
     </test-fixture>
 
     <script>
+      function waitForIronItemsChanged(selector, callback) {
```
Comment (e111077) [inline test/iron-menu-behavior.html:160]: Why is this necessary? Won't `forceSynchronousItemUpdate` work?

File: test/iron-menu-behavior.html [hunk 3]
```diff
@@ -139,107 +139,129 @@
...
-          // Wait for async focus
-          Polymer.Base.async(function() {
-            // Key press down
-            MockInteractions.keyDownOn(menu, 40);
-
+          waitForIronItemsChanged(menu, function() {
+            MockInteractions.focus(menu);
+            // Wait for async focus
             Polymer.Base.async(function() {
-              var focusedItem = Polymer.dom(menu).node.focusedItem;
```
Comment (e111077) [inline test/iron-menu-behavior.html:209]: Why remove Polymer.dom?

File: test/iron-menu-behavior.html [hunk 4]
```diff
@@ -139,107 +139,129 @@
...
```
Comment (e111077) [inline test/iron-menu-behavior.html:226]: Same as above

File: test/iron-menu-behavior.html [hunk 5]
```diff
@@ -139,107 +139,129 @@
...
           var menu = fixture('empty');
+
           MockInteractions.focus(menu);
           // Wait for async focus
           Polymer.Base.async(function() {
             // Key press down
             MockInteractions.keyDownOn(menu, 40);
 
             Polymer.Base.async(function() {
-              var focusedItem = Polymer.dom(menu).node.focusedItem;
```
Comment (e111077) [inline test/iron-menu-behavior.html:260]: This seems to happen a bunch of times. I'll just wait for a response before adding more comments

File: test/iron-menu-behavior.html [hunk 6]
```diff
@@ -333,177 +365,197 @@
...
-            assert.equal(menuActiveElement, menu.extraContent, 'menu.extraContent is focused');
-            assert.equal(Polymer.dom(document).activeElement, menu, 'menu is document.activeElement');
-            done();
+          waitForIronItemsChanged(menu, function() {
+            menu.extraContent.focus();
+            Polymer.Base.async(function() {
+              var menuOwnerRoot = Polymer.dom(menu.extraContent).getOwnerRoot() || document;
+              var menuActiveElement = Polymer.dom(menuOwnerRoot).activeElement;
+              assert.equal(menuActiveElement, menu.extraContent, 'menu.extraContent is focused');
+              assert.equal(document._activeElement || Polymer.dom(document).activeElement, menu, 'menu is document.activeElement');
```
Comment (e111077) [inline test/iron-menu-behavior.html]: can you add a comment here for `_activeElement` vs `dom().activeElement`

**Review by e111077 — CHANGES_REQUESTED** [review]

File: test/iron-menu-behavior.html (hunk 2 above)
Comment (bicknellr) [reply test/iron-menu-behavior.html:160]: Once again, this is about [this thread](https://github.com/PolymerElements/iron-selector/pull/139#discussion_r109078246) in iron-selector. :(

File: demo/simple-menu.html (hunk 1 above)
Comment (bicknellr) [reply demo/simple-menu.html]: I think I did this because I thought it would be more consistent with what `simple-menubar` is doing [here](https://github.com/PolymerElements/iron-menu-behavior/pull/78/files#diff-b26b66e5efec6300eaa47f3ef305e41fR18). I can remove it if you think that would be better?

Comment (e111077) [reply demo/simple-menu.html]: Oh it's fine. It was moved a few lines and the git diff confused me a bit

File: test/iron-menu-behavior.html (hunk 2 above)
Comment (e111077) [reply test/iron-menu-behavior.html:160]: 👍

File: test/iron-menu-behavior.html (hunk 3 above)
Comment (bicknellr) [reply test/iron-menu-behavior.html:209]: `Polymer.dom(x).node` returns `x` [in 2.x](https://github.com/Polymer/polymer/blob/master/lib/legacy/polymer.dom.html#L42) and returns a wrapper around `x` in 1.x, but these tests never used `focusedItem` to manipulate the DOM, so I removed it. I don't really know what the original intent was but I don't think there's any harm in putting it back if that's preferable.

File: test/iron-menu-behavior.html (hunk 4 above)
Comment (bicknellr) [reply test/iron-menu-behavior.html:226]: ?

File: test/iron-menu-behavior.html (hunk 5 above)
Comment (bicknellr) [reply test/iron-menu-behavior.html:260]: ([replied above](https://github.com/PolymerElements/iron-menu-behavior/pull/78/files#r110797419))

File: test/iron-menu-behavior.html (hunk 6 above)
Comment (bicknellr) [reply test/iron-menu-behavior.html]: Hmm, I need to take another look at this section. I think `_activeElement` might have been removed?

File: test/iron-menu-behavior.html (hunk 3 above)
Comment (e111077) [reply test/iron-menu-behavior.html:209]: Monica just shrugged at `Polymer.dom(...).node`'s existence. Safe to delete

File: test/iron-menu-behavior.html (hunk 4 above)
Comment (e111077) [reply test/iron-menu-behavior.html:226]: ignore request

File: test/iron-menu-behavior.html (hunk 5 above)
Comment (e111077) [reply test/iron-menu-behavior.html:260]: 👍

**Comment by bicknellr** [comment]
I need to merge master again to get changes from v1.3.0 . Depends on #91 though.

**Comment by e111077** [comment]
@bicknellr feel free to assign me to that PR

File: demo/simple-menu.html (hunk 1 above)
Comment (notwaldorf) [reply demo/simple-menu.html]: doesn't the shim require something to the left of `slotted`?

File: test/iron-menu-behavior.html (hunk 3 above)
Comment (notwaldorf) [reply test/iron-menu-behavior.html:209]: I been shrugging a lot tbh 🤷‍♀️

**Comment by bicknellr** [comment]
(^ #92 is the PR with a fix for #91.)

**Comment by bicknellr** [comment]
Ok, #92 is merged into master and master merged into 2.0-preview.

**Comment by e111077** [comment]
@bicknellr cool, any new info on `_activeElement`?

**Comment by bicknellr** [comment]
@e111077, yup, `_activeElement` doesn't seem necessary anymore: wrapping w/ `Polymer.dom` should make `activeElement` work in 1.x & 2.x:
1.x (Shady DOM) https://github.com/Polymer/polymer/blob/afb233ce812b2b4971ab9aa51a8fdff96a571441/src/lib/dom-api-shady.html#L458
1.x (Shadow DOM) https://github.com/Polymer/polymer/blob/afb233ce812b2b4971ab9aa51a8fdff96a571441/src/lib/dom-api-shadow.html#L61
2.x https://github.com/Polymer/polymer/blob/1478a0680ab822e786eee0ad93c9d8937bff31d4/lib/legacy/polymer.dom.html#L177

**Review by e111077 — APPROVED** [review]

File: demo/simple-menu.html (hunk 1 above)
Comment (bicknellr) [reply demo/simple-menu.html]: [Yup](https://github.com/webcomponents/shadycss#selector-scoping), I'll push a fix.

Comment (bicknellr) [reply demo/simple-menu.html]: pushed
